background.js

```diff
@@ -120,14 +117,28 @@ ipcMain.on('k8s-restart', () => {
     k8smanager.stop()
       .then(() => {
         // The desired Kubernetes version might have changed
-        k8smanager = K8s.factory(cfg.kubernetes);
+        k8smanager = newK8sManager(cfg.kubernetes);
       })
-      .then(() => { k8smanager.start() })
+      .then(() => { return k8smanager.start() })
       .then(() => {
+        try {
+          event.reply('k8s-check-state', k8smanager.state);
+        } catch (err) {
+          console.log(err);
+        }
       }, startfailed);
   }
 })
 
 function startfailed(code) {
   dialog.showErrorBox("Error Starting Kuberentes", "Kubernetes was unable to start with the following exit code: " + code);
+}
+
+function newK8sManager(cfg) {
```

Comment:
Shouldn't we change line 91 to use this too?  When I read the description I thought this was what you meant.  I was dumb and didn't re-hook-up the event listener when making a new instance.

Reply:
Good catch. I missed that one.